app/code/Magento/CustomerGraphQl/Model/Resolver/Address.php

```diff
@@ -0,0 +1,278 @@
+use Magento\Eav\Model\Config;
+use Magento\Framework\GraphQl\Exception\GraphQlAuthorizationException;
+use Magento\Framework\GraphQl\Exception\GraphQlNoSuchEntityException;
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+use Magento\Framework\Exception\NoSuchEntityException;
+
+/**
+ * Customers Address, used for GraphQL request processing.
+ */
+class Address implements ResolverInterface
+{
+    /**
+     * Mutation Address type
+     */
+    const MUTATION_ADDRESS_CREATE = 'customerAddressCreate';
+    const MUTATION_ADDRESS_UPDATE = 'customerAddressUpdate';
+    const MUTATION_ADDRESS_DELETE = 'customerAddressDelete';
+
+    /**
+     * @var AddressRepositoryInterface
+     */
+    private $addressRepositoryInterface;
+
+    /**
+     * @var AddressInterfaceFactory
+     */
+    private $addressInterfaceFactory;
+
+    /**
+     * @var Config
+     */
+    private $eavConfig;
+
+    /**
+     * @var AddressDataProvider
+     */
+    private $addressDataProvider;
+
+    /**
+     * @var DataObjectHelper
+     */
+    private $dataObjectHelper;
+
+    /**
+     * @var array
+     */
+    private $addressAttributes;
+
+    /**
+     * @param AddressRepositoryInterface $addressRepositoryInterface
+     * @param AddressInterfaceFactory $addressInterfaceFactory
+     * @param Config $eavConfig
+     * @param AddressDataProvider $addressDataProvider
+     * @param DataObjectHelper $dataObjectHelper
+     */
+    public function __construct(
+        AddressRepositoryInterface $addressRepositoryInterface,
+        AddressInterfaceFactory $addressInterfaceFactory,
+        Config $eavConfig,
+        AddressDataProvider $addressDataProvider,
+        DataObjectHelper $dataObjectHelper
+    ) {
+        $this->addressRepositoryInterface = $addressRepositoryInterface;
+        $this->addressInterfaceFactory = $addressInterfaceFactory;
+        $this->eavConfig = $eavConfig;
+        $this->addressDataProvider = $addressDataProvider;
+        $this->dataObjectHelper = $dataObjectHelper;
+        $this->addressAttributes = $this->eavConfig->getEntityAttributes(
+            AddressMetadataManagementInterface::ENTITY_TYPE_ADDRESS
+        );
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        /** @var \Magento\Framework\GraphQl\Query\Resolver\ContextInterface $context */
+        if ((!$context->getUserId()) || $context->getUserType() == UserContextInterface::USER_TYPE_GUEST) {
+            throw new GraphQlAuthorizationException(
+                __(
+                    'Current customer does not have access to the resource "%1"',
+                    [AddressMetadataManagementInterface::ENTITY_TYPE_ADDRESS]
+                )
+            );
+        }
+        $customerId = $context->getUserId();
+        switch ($field->getName()) {
```

Comment:
Pls, split this class into three different classes `mutation resolvers`

Reply:
Refactor done! :)